falcon/asgi/request.py

```diff
@@ -321,14 +321,27 @@ class Request(request.Request):
         if_range (str): Value of the If-Range header, or ``None`` if the
             header is missing.
 
-        headers (dict): Raw HTTP headers from the request with
-            canonical dash-separated names. Parsing all the headers
-            to create this dict is done the first time this attribute
-            is accessed, and the returned object should be treated as
-            read-only. Note that this parsing can be costly, so unless you
-            need all the headers in this format, you should instead use the
-            ``get_header()`` method or one of the convenience attributes
-            to get a value for a specific header.
+        headers (dict): Raw HTTP headers from the request with dash-separated
+            names normalized to lowercase.
+
+            Note:
+                This property differs from the WSGI version of ``Request.headers``
+                in that the latter returns *uppercased* names for historical
+                reasons. Middleware, such as APM and logging components, that
```

Comment:
I may be out of the loop here, but it's not clear to me what APM is

Reply:
ah, application performance monitoring (e.g., ddtrace). Maybe better to `:s/APM/tracing/g`?

Reply:
done

Reply:
tracing is clearer (then again maybe I'm the one OOTL)